lessons/lesson5.md

```diff
@@ -0,0 +1,196 @@
+
+Each icon has to be imported individually with a single statement.
+
+```javascript
+import TaskList from '@spectrum-icons/workflow/TaskList';
+import Close from '@spectrum-icons/workflow/Close';
+import Add from '@spectrum-icons/workflow/Add';
+```
+
+A [searchable list of workflow icons](https://spectrum.adobe.com/page/icons/) is available on the Spectrum website. The name of the icon without any whitespace matches the import in React Spectrum.  
+
+### Displaying the todo list name
+
+React Spectrum has made available a set of components and keeps delivering new components with every release. Unfortunately, some components are not available yet in React Spectrum but available in [Spectrum CSS](https://opensource.adobe.com/spectrum-css/) which is the CSS framework used by React Spectrum to style components. 
+Luckily for us, we can use React Spectrum together with Spectrum CSS to fill the gap of missing components.
+
+In this case, we're going to install the [Spectrum CSS Typography](https://opensource.adobe.com/spectrum-css/typography.html) dependency to render Spectrum headings.
+Since React Spectrum already defines the Spectrum CSS variables, you'll only need to install the Spectrum CSS Typography package.
+
+```bash
+npm i --save @spectrum-css/typography
+``` 
+
+Then you can simply import the CSS with:
+
+```javascript
+import '@spectrum-css/typography';
+``` 
+
+Now you can use the Spectrum CSS Heading classes to render the todo list name.
+
+```jsx
+<h2 className="spectrum-Heading spectrum-Heading--sizeM spectrum-Heading--serif">{name}</h2>
+```
+
+### Rendering todo items
+
+Once again, we're going to use the [State hook](https://reactjs.org/docs/hooks-state.html) `useState` to declare a state variable which will hold a list of todo items.
+This list will be updated whenever a new todo item is created. By default, it's initialized with the `todos` from the `todoList` prop. 
+
+```javascript
+const [todoItems, setTodoItems] = useState(todos);
+```
+
+Next we'll iterate over the `todoItems` array using the [map()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map) function to render each item as `Todo` component.
+
+```jsx
+<View marginTop="size-100">
+  {todoItems.map((todo) => (
+    <Todo key={todo.id} name={name} todo={todo} onUpdate={onUpdate} />
+  ))}
+</View>
+```
+
+The `key` property is necessary in React to uniquely identify the todo item. In this case, we use the todo id. 
+You can read more about React lists and keys [here](https://reactjs.org/docs/lists-and-keys.html).
+
+We're also passing the `name`, `todo` and `onUpdate` props down to the `Todo` component.      
+
+### Creating a todo item
+
+If you remember in [lesson 2](lesson2.md), we defined a `MAX_TODO_ITEMS` value within a global configuration file `defaults.json` at the root of the Firefly App.
```

Comment:
As it comes to here, the values in `defaults.json` file are shared between server and client. This is not recommended because the two sides run on different machines, and create complications at build and deploy (and this is something we don't promote in Firefly code).
Ideally, the value should be set for and retrieved by the action, then the UI based on the payload to determine how it renders the list of items. An additional attribute `size` in the response payload might help here.

Reply:
Ok thanks I didn't know. Do you have more information on this topic ?

> Ideally, the value should be set for and retrieved by the action, then the UI based on the payload to determine how it renders the list of items

Ok then I'll follow that path.

Reply:
more info about this topic here: https://github.com/AdobeDocs/project-firefly/blob/master/guides/application_state.md

Reply:
Thanks, based on the docs and internal discussions, I'll leave it for now until there's a Firefly defined way to do it eg inside `app.config` file as you mentioned.